iceoryx_posh/source/mepoo/chunk_settings.cpp

```diff
@@ -63,41 +63,50 @@ expected<ChunkSettings, ChunkSettings::Error> ChunkSettings::create(const uint32
         return err(ChunkSettings::Error::USER_HEADER_SIZE_NOT_MULTIPLE_OF_ITS_ALIGNMENT);
     }
 
-    uint64_t requiredChunkSize =
-        calculateRequiredChunkSize(userPayloadSize, adjustedUserPayloadAlignment, userHeaderSize);
-
-    if (requiredChunkSize > std::numeric_limits<uint32_t>::max())
+    auto expectChunkSize = calculateRequiredChunkSize(userPayloadSize, adjustedUserPayloadAlignment, userHeaderSize);
+    if (expectChunkSize.has_error())
     {
-        return err(ChunkSettings::Error::REQUIRED_CHUNK_SIZE_EXCEEDS_MAX_CHUNK_SIZE);
+        return err(expectChunkSize.get_error());
     }
+    uint64_t requiredChunkSize = expectChunkSize.value();
 
-    return ok(ChunkSettings{userPayloadSize,
-                            adjustedUserPayloadAlignment,
-                            userHeaderSize,
-                            adjustedUserHeaderAlignment,
-                            static_cast<uint32_t>(requiredChunkSize)});
+
+    return ok(ChunkSettings{
+        userPayloadSize, adjustedUserPayloadAlignment, userHeaderSize, adjustedUserHeaderAlignment, requiredChunkSize});
 }
-uint64_t ChunkSettings::calculateRequiredChunkSize(const uint32_t userPayloadSize,
-                                                   const uint32_t userPayloadAlignment,
-                                                   const uint32_t userHeaderSize) noexcept
+
+expected<uint64_t, ChunkSettings::Error> ChunkSettings::calculateRequiredChunkSize(
+    const uint64_t userPayloadSize, const uint32_t userPayloadAlignment, const uint32_t userHeaderSize) noexcept
 {
     // have a look at »Required Chunk Size Calculation« in chunk_header.md for more details regarding the calculation
     if (userHeaderSize == 0)
     {
         // the most simple case with no user-header and the user-payload adjacent to the ChunkHeader
         if (userPayloadAlignment <= alignof(mepoo::ChunkHeader))
         {
+            if (userPayloadSize > std::numeric_limits<uint64_t>::max() - sizeof(ChunkHeader))
+            {
+                return err(ChunkSettings::Error::REQUIRED_CHUNK_SIZE_EXCEEDS_MAX_CHUNK_SIZE);
+            }
```

Comment:
Since you've added this (unlikely) error case, it'd probably be good to add a unit test as well that calls the function with exactly the bad threshold size and asserts that an error is returned.

Reply:
Same for the other error case.

Reply:
If I did not overlook something, there are already test cases for all of the possible errors. It just happend that the checks are now partially at a different location. At least the coverage of these files is 100%

Reply:
Ah okay. I guess I didn't look hard enough and assumed because there wasn't a new unit test calling this function directly and exercising the expected value, it might not be checked. 

But you are right - this is the same error that existed before.